### ISSUE TITLE
Allow addition of empty 2d grid for runtime manipulation, like 1d grid demonstrated in tutorials

### ISSUE THREAD

**Issue by Levitikon217** [goncasmage1/UINavigation#28]
Implementing a real dynamic inventory system involves 2d grids and runtime manipulation, which doesn't seem to be currently supported. I removed all the parameter count checks in AppendNavigationGrid2D and dropped in `UINavButtons.Num() > 0 ? UINavButtons[NumberOfButtonsInGrids] : nullptr` in 2nd param of FGrid ctor, and all seems to work fine. I suspect you'll still want to validate the parameters in some way, but the point is, an empty 2d grid that is populated at runtime should be allowed and verified works fine.

**Comment by goncasmage1** [comment]
What version of the plugin are you currently using?

A sábado, 21/09/2019, 16:41, Levitikon <notifications@github.com> escreveu:

**Comment by Levitikon217** [comment]
UE 4.22

**Comment by goncasmage1** [comment]
When you open the Plugins window and go to UI Navigation, what version
shows up?

A sábado, 21/09/2019, 20:39, Levitikon <notifications@github.com> escreveu:

**Comment by Levitikon217** [comment]
2.7.2
I pulled it in through UE marketplace this morning.

**Comment by goncasmage1** [comment]
How are you creating the grids? Can you send a screenshot?

**Comment by Levitikon217** [comment]
Yeah sure. For what it's worth, this is a pretty slick plugin you have here.

I'm still in the super early stages of putting it all together, so screenshots illustrate a hard-coded set of inventory icons in 3 groups and 8 icons each, but at some point this will be entirely dynamic based on a character that can have an un-bounded set of items that will grouped by category (swords, axes, bows, etc). So each group (grid) needs to be a wrap box that is sized so that only 5 icons can fit before wrapping. This aligns with the Dimension X of the 2d grid. So it's:
```
<verticle box>
   <header>
   <hr>
   <wrap box>
      <nav button>
      <nav button>
      ....
   </wrap box>
   <header>
   <hr>
   <wrap box>
      <nav button>
      <nav button>
      ....
   </wrap box>
   ...repeat...
</verticle box>
```

This actually works out very well. This only thing is, I had to adjust the c++ code to make it work. I found the original 0 button parameter enforcement unnecessary. This works just fine:
```
void UUINavWidget::AppendNavigationGrid2D(int DimensionX, int DimensionY, FButtonNavigation EdgeNavigation, bool bWrap, int ButtonsInGrid)
{
	FButtonNavigation NewNav;
	FGrid NewGrid = FGrid(EGridType::Grid2D, UINavButtons.Num() > 0 ? UINavButtons[NumberOfButtonsInGrids] : nullptr, NavigationGrids.Num(), DimensionX, DimensionY, EdgeNavigation, bWrap, ButtonsInGrid);
	...
}
```

![Capture](https://user-images.githubusercontent.com/1824102/65379676-bbc70480-dc99-11e9-8e58-7ad7a2f20a82.PNG)
![Capture2](https://user-images.githubusercontent.com/1824102/65379677-bbc70480-dc99-11e9-992e-fbffcd6515f3.PNG)
![Capture3](https://user-images.githubusercontent.com/1824102/65379678-bbc70480-dc99-11e9-8639-251cc1cd1dba.PNG)
